core_backend/tests/api/test_key_management.py

```diff
@@ -0,0 +1,17 @@
+from fastapi.testclient import TestClient
+
+from core_backend.tests.api.conftest import TEST_USER_RETRIEVAL_KEY_2
+
+
+class TestKeyManagement:
+    async def test_get_new_retrieval_key(
```

Comment:
This is my first time reviewing multiuser feature.. so sorry if I missed some context here, but is this key just for /embeddings-search? Or also for llm generated responses & UD? Thought "retrieval" would be confusing if we can do all three operations with the same key!

Reply:
Yes you're right, when I renamed it to that we didn't have UD yet. Will change it to api_key in a separate final PR before merging mulittenancy to main, thanks!